src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -107,6 +229,39 @@ InvokerFactory MakeImplGemmDynamicDataInvokerFactory(const ConvolutionContext& c
             };
         };
     }
+    else if(ctx.direction.IsBackwardData())
+    {
+        return [ctx](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const boost::any& primitive_parameters) {
+                const auto data_ctx = boost::any_cast<conv::DataInvokeParams>(primitive_parameters);
+                const auto& tensors = data_ctx.tensors;
+                auto kernel         = handle.Run(kernels[0]);
+                if(kernel.GetName().find("igemm_bwd_gtc") == 0)
+                {
+                    std::vector<KernelInvoke> ks;
+                    std::transform(kernels.begin(),
+                                   kernels.end(),
+                                   std::back_inserter(ks),
+                                   [&](const Kernel& k) { return handle.Run(k); });
+                    float elapsed = 0;
+
+                    elapsed = CallImplicitGemmDynamic(
+                        handle, ctx, tensors.in, tensors.out, tensors.w, ks);
+
+                    if(handle.IsProfilingEnabled())
+                    {
+                        handle.ResetKernelTime();
+                        handle.AccumKernelTime(elapsed);
+                    }
+                }
+                else
+                {
+                    MIOPEN_THROW(
+                        "Error running dynamic implicit GEMM convolution (invalid kernel name?)");
```

Comment:
When you throw error, you could add kernel name in the string so one knows which kernel name is problematic.

Reply:
This whole file should be split to separate invokers, and no kernel name-checks should be present anywhere in the library. But for now, it is a wise suggestion, I guess.

Reply:
I think if no kernel name check should be present, there might be a lot of re-factor work (even in HIP based igemm), so I guess current implementation is OK?

Reply:
@carlushuang Please add kernel name in throw message at bare minimum

Reply:
OK, let me add it

Reply:
Unfortunately, this is not ok.

You do not need to refactor all igemm invokers, but the new code should be written properly.

Reply:
Hi @atamazov I've seperate the invoker in different path (fwd/bwd...)